db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/subquery/exists/ExistsSubqueryFlatteningVisitor.java

```diff
@@ -381,12 +397,60 @@ private FromSubquery addFromSubquery(SelectNode outerSelectNode,
             /*
              * Insert the new FromSubquery into to origSelectNode's From list.
              */
-            outerSelectNode.getFromList().getNodes().clear();
+            // For cases with multiple items in the FROM clause, remove the appropriate
+            // table, otherwise just clear the FROM list.
+            if (outerFromTable == null) {
+                outerSelectNode.getFromList().getNodes().clear();
+            }
+            else {
+                int originalSize = outerSelectNode.getFromList().size();
+                outerSelectNode.getFromList().removeElement(outerFromTable);
+                if (outerSelectNode.getFromList().size() == originalSize)
+                    throw StandardException.newException(LANG_INTERNAL_ERROR,
```

Comment:
the same question as above

Reply:
Yes, this is defensive.   If for whatever reason we could not remove the table from the FROM list, that means perhaps that outerFromTable is pointing to an object not in the FROM list, which should never happen.  We want to catch this case if it does happen, otherwise the query would return incorrect results.